### DISCUSSION TITLE
Hacktribe bricked e2 help please

### DISCUSSION THREAD

**Discussion by Obuzus** [bangcorrupt/hacktribe#66] [Q&A] [answered]
Please help with bricked e2 after hacktribe update, coud anyone explain in detail how debrick-ing device is done?

**Comment by bangcorrupt** [comment, marked as answer]
Hi, I can create a more approachable de-brick guide. It may take a while, I'm miles behind with other work.  

I've seen a few mentions of bricking during update, I'm interested to find what led to this.  Do you know what caused the problem?

**Reply by Obuzus** [reply]
I wa unable to format the card (card size error) and ignored that part :( I purchased to replaced the system file with the one created from terminal...

**Reply by Obuzus** [reply]
Done, thank you very much for your time :)

**Reply by bangcorrupt** [reply]
It looks like you applied the patch to the synth version of the firmware instead of the sampler version.  Hacktribe patch is a modification to the sampler firmware, which will run on either device (the only difference between them is the data stored in the flash).  This is what the first hash check is for, see [FAQ](https://github.com/bangcorrupt/hacktribe/wiki/FAQ#what-the-hash-is-a-checksum).  

~~I've tried to make this less error prone with an [automated patching script](https://github.com/bangcorrupt/e2-scripts/blob/master/e2-firmware-patch.py), see [Patch Automatically](https://github.com/bangcorrupt/hacktribe/wiki/How-To#patch-automatically) in the wiki for more details.~~

In the future I'd like to hack the bootloader so de-bricking doesn't require opening the case.  I'll try to put together a guide with screenshots and photos over the next couple of weeks.  Good news is, this can be fixed and your data is safe :).

EDIT:
Follow the [Hacktribe Tutorial](https://github.com/bangcorrupt/hacktribe-tutorial) to get the patched firmware files.

**Reply by Obuzus** [reply]
instead to patch the fw for synth i patched and applied the file system for sampler? I understand corect?

**Reply by bangcorrupt** [reply]
Other way round.  You patched the synth firmware.  You should have patched the sampler firmware, then installed it on the synth.

**Reply by Obuzus** [reply]
Now i understand, thank you very much for info.

**Reply by Obuzus** [reply]
Is posibile to use rpi zero 2w with openocd to debrick the e2? Thanks :)

**Reply by bangcorrupt** [reply]
Yes, that will work.  Doesn't need to be version 2.  See [this thread](https://github.com/bangcorrupt/hacktribe/discussions/79) for a few more details.
